docs/operator-bundle/testing-and-releasing.md

```diff
@@ -0,0 +1,181 @@
+```
+
+The previous version of the operator will install and start up. You can check progress by looking at the yaml for the install plan and [ClusterServiceVersion](https://docs.openshift.com/container-platform/4.7/rest_api/operatorhub_apis/installplan-operators-coreos-com-v1alpha1.html), the logs from the catalog operator, and eventually the state of the pods in the `operators` namespace.
+
+### Approve installation of the new version
+
+Once the old version of the operator is installed, the OLM will create another install plan to upgrade it to the new version.
+Approve that plan in the same way as the previous one and watch progress.
+
+### Troubleshooting
+
+If there's an issue with the new bundle, it will probably show up as a condition on the install plan (as well as logging by the catalog operator).
+Problems I've seen in the past include:
+
+* Validation errors in the CSV (particularly the `webhookdefinitions` section).
+* Validation errors when applying the CRDs (eg. incorrect conversion fields after the v1beta1 -> v1 switch)
+* Incorrect container image reference in deployment - this manifested as the install plan stuck in Installing (no errors) but the deployed pods stuck failing to pull the image.
+
+To back out the installed operator and retry:
+
+1. Fix the error and push a new bundle image as above - be sure to push the bundle image to the same version tag since that's what the index refers to.
+2. Delete the subscription (will also remove the install plans):
+    ```sh
+    k delete -n operators subscriptions my-azure-service-operator
+    ```
+3. Delete the CSVs (this will also remove any deployments/services/pods etc.):
+    ```sh
+    k delete -n operators csv azure-service-operator.v<old> azure-service-operator.v<new>
+    ```
+4. Remove the CRDs:
+    ```sh
+    k api-resources --api-group azure.microsoft.com -o name | xargs kubectl delete customresourcedefinition
+    ```
+5. The catalog operator creates a configmap of each bundle image to load the CSV and CRDs from.
+This won't be recreated from the updated image if it already exists (even though the content is out of date), so you need to delete it.
+They're named with hashes but you can look at the `olm.sourceImage` annotation to find the right one to delete.
+    ```sh
+    k get -n olm configmap -o jsonpath='{range .items[*]}{@.metadata.name} {@.metadata.annotations.olm\.sourceImage}{"\n"}{end}'
+    k delete -n olm configmap <selected-hash>
+    ```
+6. Then recreating the subscription as above will restart the install/upgrade process.
+
+## Creating PRs to add the new version to the [community-operators](https://github.com/operator-framework/community-operators) repo
+
+The embedded OpenShift operator store and [OperatorHub](https://operatorhub.io) are both populated from the community-operators repository.
+The OpenShift store bundles are under the `community-operators` directory, while OperatorHub ones are under `upstream-community-operators`.
```

Comment:
minor: These could be direct links to the directories in GitHub if you wanted. Not a big deal either way though